flink-python/pyflink/testing/source_sink_utils.py

```diff
@@ -0,0 +1,122 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+#################################################################################
+
+import glob
+import os
+import unittest
+from py4j.java_gateway import java_import
+
+from pyflink.find_flink_home import _find_flink_source_root
+from pyflink.java_gateway import get_gateway
+from pyflink.table import TableSink
+
+
+class TestTableSink(TableSink):
+    """
+    Base class for test table sink.
+    """
+
+    def __init__(self, j_table_sink_name, j_table_sink):
+        FLINK_SOURCE_ROOT_DIR = _find_flink_source_root()
+        filename_pattern = (
+            "flink-table/flink-table-planner/target/"
+            "flink-table-planner*-tests.jar")
+        if not glob.glob(os.path.join(FLINK_SOURCE_ROOT_DIR, filename_pattern)):
```

Comment:
Should we check this in a more appropriate position such as the setUp() method of the test case or launch_gateway()?

Reply:
make sense. updated!